transaction-view/src/resolved_transaction_view.rs

```diff
@@ -0,0 +1,570 @@
+use {
+    crate::{
+        result::{Result, TransactionViewError},
+        transaction_data::TransactionData,
+        transaction_view::TransactionView,
+    },
+    core::{
+        fmt::{Debug, Formatter},
+        ops::Deref,
+    },
+    solana_sdk::{
+        bpf_loader_upgradeable, ed25519_program,
+        hash::Hash,
+        message::{v0::LoadedAddresses, AccountKeys, TransactionSignatureDetails},
+        pubkey::Pubkey,
+        secp256k1_program,
+    },
+    solana_svm_transaction::{
+        instruction::SVMInstruction, message_address_table_lookup::SVMMessageAddressTableLookup,
+        svm_message::SVMMessage,
+    },
+    std::collections::HashSet,
+};
+
+/// A parsed and sanitized transaction view that has had all address lookups
+/// resolved.
+pub struct ResolvedTransactionView<D: TransactionData> {
+    /// The parsed and sanitized transction view.
+    view: TransactionView<true, D>,
+    /// The resolved address lookups.
+    resolved_addresses: Option<LoadedAddresses>,
+    /// A cache for whether an address is writable.
+    writable_cache: Vec<bool>, // TODO: should this be a vec, bitset, or array[256].
+}
+
+impl<D: TransactionData> Deref for ResolvedTransactionView<D> {
+    type Target = TransactionView<true, D>;
+
+    fn deref(&self) -> &Self::Target {
+        &self.view
+    }
+}
+
+impl<D: TransactionData> ResolvedTransactionView<D> {
+    /// Given a parsed and sanitized transaction view, and a set of resolved
+    /// addresses, create a resolved transaction view.
+    pub fn try_new(
+        view: TransactionView<true, D>,
+        resolved_addresses: Option<LoadedAddresses>,
+        reserved_account_keys: &HashSet<Pubkey>, // why does this not use ahash at least?!
```

Comment:
🙃

Reply:
b387ac962efcdf764647961719ba7105f0fe9eb5

actually removed. i swear i thought i did this before!